### PR TITLE
Deploy prototype DSA server

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
**Important:** This PR depends on #1 (to ensure that the VPC is created).

In this PR, I recreate the [stack](https://github.com/Sage-Bionetworks/sandbox-provisioner/blob/master/config/prod/dgutman-htan-new2-ec2.yaml) deployed in `sandbox` for hosting the Digital Slide Archive (DSA). Our collaborators maintain the server via SSH, so we thought it would be better to host the instance in a separate AWS account.

### CODE REVIEW

**Comment by BrunoGrandePhD** [comment]
Manually requesting a review from @ahayden since he hasn't accepted the repo invitation yet.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:21]: these params are specific to sandbox and scicomp

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:18]: @ahayden should review your intention here.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:27]: this is specific to sandbox and scicomp

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:17]: vpc is setup in a specific say on sandbox and scicomp, i'm not sure if it's compatible with this VPC setup.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:3]: i suggest you don't use this template since it's sorta specific to sandbox and scicomp.  You can either create a custom and put it in this repo or create a more generic one and put it in aws-infra repo to share.

**Review by zaro0508 — CHANGES_REQUESTED** [review]
blank

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (milen-sage) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:17]: @zaro0508 what are the VPC parameters compatible w/ sandbox?

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (milen-sage) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:3]: @zaro0508 would the template in this PR be sufficient provided we are spinning a sandbox instance? I.e. what are the advantages of a custom or generic templates over the one here?

Comment (ahayden) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:3]: I think it's fine to use this template... will also want the other sandbox infrastructure in this account, like maintenance window, quarantine security group, etc.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (ahayden) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:18]: It looks to me like this server is used for production work (despite it being called dev), in which case it should not be in a public subnet. This is a very fragile deployment and is a bad practice to use a single host as web server and jumpbox.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (ahayden) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:19]: This should not be a jumpcloud connected host. AFAIK dgutman will have an SSH key pair outside of JC. In prod, we would not deploy a JC-connected web server not use the web server as a jumpbox.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (ahayden) [inline config/dev/dgutman-dsa-server-preprod-ec2.yaml:8]: Port 80 should not be only on this instance. If this is a web app that will be used by anyone except developers, it should have an ELB in front and the instance should not have to do any 80->443 redirect. If it truly is just for dev, this is probably ok to open these ports, but it should not be considered secure. IOW, this is not a configuration we would replicate in prod.

**Review by ahayden — CHANGES_REQUESTED** [review]

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml]: I've gotten rid of these params.

**Comment by BrunoGrandePhD** [comment]
I've captured the various expectations from IT that have been mentioned so far on this PR on the [JIRA ticket](https://sagebionetworks.jira.com/browse/IT-1214?focusedCommentId=130370) tracking the long-term production deployment.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:8]: I've noted this requirement for the production template [here](https://sagebionetworks.jira.com/browse/IT-1214?focusedCommentId=130370).

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:18]: If HTAN can tolerate the fragility, does this block the PR?

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:17]: @zaro0508: Where is this documented? I thought templates in `aws-infra` were general-purpose. Could you clarify what exactly you expect to cause issues?

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:3]: Is this "sandbox-like infrastructure" a requirement for this pre-production server? Or can it wait until the production deployment? This is the first time that I hear about these additional requirements for new AWS accounts and would have expected these to be deployed automatically by org-formation if they're even required on dev accounts.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (zaro0508) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:17]: it's not necessarily `general purpose`, it's any template that you want to share and deploy to multiple accounts.  those files were used in the two accounts (sandbox & scicomp) that were setup for git based provisioning.  that template contains a lot more than you need for your purpose that's why i suggest you use your own template or just pair that template down into something much simpler.   One thing that will break is referenced to essentials which you do not have in your account. https://github.com/Sage-Bionetworks/aws-infra/blob/v0.2.13/templates/managed-ec2-ubuntu-v2.j2#L241

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (ahayden) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:3]: Yes, I agree that it should be deployed for org-f or sceptre for all member accounts. It shouldn't require any extra set up

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (ahayden) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:18]: If it's a prototype, I don't have a problem with it... HTAN does not get to define Sage's security practices... The project is either a prototype, in which is is not used for production. Or, is something that we will support as production work, in which case it needs to be designed according to IT Ops standards.

**Review by ahayden — APPROVED** [review]
Approving changes for use as prototype. There should be a definite end date for when we turn off the resources in this account.

**Comment by BrunoGrandePhD** [comment]
I've created a calendar event in three months (September 1st) as a reminder to take down this prototype server.

File: config/dev/dgutman-dsa-server-preprod-ec2.yaml
Comment (BrunoGrandePhD) [reply config/dev/dgutman-dsa-server-preprod-ec2.yaml:17]: I've added a copy of the `managed-ec2-linux-v2.j2` template to this repo directly on `main`. This way, any edits that I made to the template in this branch will come up as diffs. Note that I copied the instance role from `essentials.yaml`.

**Comment by BrunoGrandePhD** [comment]
@zaro0508: Can you make sure that my edited version of `managed-ec2-linux-v2.j2` looks good? Thanks!

**Comment by zaro0508** [comment]
> @zaro0508: Can you make sure that my edited version of `managed-ec2-linux-v2.j2` looks good? Thanks!

You may want to consider using template in PR https://github.com/Sage-Bionetworks/aws-infra/pull/306 instead?

**Comment by BrunoGrandePhD** [comment]
@zaro0508: I've approved Sage-Bionetworks/aws-infra#306, but for this PR, we'll need a way to grant external collaborators with shell access. I've made the request [here](https://github.com/Sage-Bionetworks/aws-infra/pull/306#pullrequestreview-690207082).

File: config/test/dsa-prototype-ec2.yaml
Comment (zaro0508) [inline config/test/dsa-prototype-ec2.yaml:10]: default for this is `true` so probably don't need to specify.

File: config/test/dsa-prototype-ec2.yaml
Comment (zaro0508) [inline config/test/dsa-prototype-ec2.yaml:6]: default instance size is `t3.nano`.  hope that's not too small for you.

**Review by zaro0508 — APPROVED** [review]

File: config/test/dsa-prototype-ec2.yaml
Comment (BrunoGrandePhD) [reply config/test/dsa-prototype-ec2.yaml:10]: Yeah, I noticed. Including this just to be explicit/self-documenting.

File: config/test/dsa-prototype-ec2.yaml
Comment (BrunoGrandePhD) [reply config/test/dsa-prototype-ec2.yaml:6]: The instance type is specified on line 7: `InstanceType: "i3en.2xlarge"`. So, this shouldn't be an issue.

**Pull request merged**
